src/resources/classes/timeSeriesList.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 Cognite AS
+import { uniqBy } from 'lodash';
+import { CogniteClient } from '../..';
+import { DatapointsMultiQuery } from '../../types/types';
+import { TimeSeries } from './timeSeries';
+
+export class TimeSeriesList extends Array<TimeSeries> {
+  private client: CogniteClient;
+  constructor(client: CogniteClient, items: TimeSeries[]) {
+    super(...items);
+    this.client = client;
+  }
+
+  /**
+   * Deletes all timeseries in the TimeSeriesList
+   * ```js
+   * await timeSeriesList.delete();
+   * ```
+   */
+  public delete = async () => {
+    return this.client.timeseries.delete(
+      this.map(timeseries => ({ id: timeseries.id }))
+    );
+  };
+
+  /**
+   * Retrieves all the assets related to all the timeseries in the TimeSeriesList
+   * ```js
+   * const assets = await timeSeriesList.getAllAssets();
+   * ```
+   */
+  public getAllAssets = async () => {
+    const assetIds = this.map(timeseries => ({
+      id: timeseries.assetId as number,
+    })).filter(assetId => assetId.id !== undefined);
+    return this.client.assets.retrieve(uniqBy(assetIds, 'id'));
```

Comment:
I am not sure if the second parameter to `uniqBy` can be a string and not a function